docs/data-hub/dataframes.md

```diff
@@ -0,0 +1,130 @@
+
+
+=== "Modify entire columns"
+
+    To modify entire columns, use native pandas syntax:
+
+    ```python
+    df["y^2"] = df["y"] ** 2
+    ```
+
+=== "Modify data in a single cell"
+
+    To modify data in a single cell, use native pandas syntax (the at operator):
+
+    ```python
+    df.at["sgs-1", "y"] = 10
+    ```
+
+In either case, when we view the DataFrame once more using `df`, we see the updated data, together with a warning telling us that we have local changes that haven't been written back to Deep Origin.
+
+
+![DataFrame with warning telling us about local changes that have not been written back to Deep Origin](../images/dataframe-1.png)
+
+## Write data back to Deep Origin
+
+### Using the `to_deeporigin` method
+
+Local changes in the dataframe can be written back to Deep Origin using the `to_deeporigin` method:
+
+```python
+df.to_deeporigin()
+
+# ✔︎ Wrote 9 rows in y^2 to Deep Origin database.
+```
+
+!!! tip "Intelligent writing"
+    - Deep Origin DataFrames keep track of local changes, and only write columns back that have been modified locally. 
+    - Every call of `to_deeporigin` will generate a print statement describing the changes that have been written back to Deep Origin. 
+    - Because a Deep Origin DataFrame corresponds to a database on Deep Origin, there is no need to specify the database name in the `to_deeporigin` method.
+
+If we now view the dataframe once more using `df`, we see the following:
+
+![DataFrame](../images/dataframe-2.png)
+
+Note that the warning about local changes that have not been written back to Deep Origin has disappeared, because the changes **have** been written back to Deep Origin.
+
+### Automatic writing to Deep Origin
+
+All Deep Origin DataFrames have an attribute called `auto_sync` that determines if local changes are written automatically to Deep Origin. By default, `auto_sync` is set to `False`, requiring you to call the `to_deeporigin` method to write changes back to Deep Origin.
+
+To enable automatic syncing, set the `auto_sync` attribute to `True`:
+
+```python
+df.auto_sync = True
+df
+```
+
+![DataFrame](../images/dataframe-3.png)
+
+Note that the dataframe now displays a message indicating that local changes will be written back to Deep Origin.
+
+Making any change to the dataframe now triggers a write back to the Deep Origin database.
+
+```python
+df["y^2"] = df["y"] * 0.99
+df
+
+# ✔︎ Wrote 9 rows in y^2 to Deep Origin database.
+```
+
+!!! danger "Use `auto_sync` with caution"
+    Turning on `auto_sync` on dataframes can be dangerous. 
+    - Changes made to the local database are written to a Deep Origin database automatically, and no confirmation is asked for. 
+    - This can cause data loss.
+    - Every change made to the database is written immediately, so modifying the local dataframe multiple times leads to multiple writes to a Deep Origin database.
+
+## Limitations
+
```

Comment:
Before stating what this doesn't do, I think it would be instructive to add 1 more sentence that says what `to_deeporigin` does do -- update the values of existing cells.

Reply:
fixed